portal-api/src/v1/gef/services/clone-facilities.service.ts

```diff
@@ -0,0 +1,61 @@
+import { ObjectId } from 'mongodb';
+import { produce } from 'immer';
+import { isBefore, startOfDay } from 'date-fns';
+import { AuditDetails, Facility } from '@ukef/dtfs2-common';
+import { generateAuditDatabaseRecordFromAuditDetails } from '@ukef/dtfs2-common/change-stream';
+import { mongoDbClient } from '../../../drivers/db-client';
+
+/**
+ * Clones the existing facilities, removing/updating fields
+ * @param currentDealId - the existing deal id
+ * @param newDealId - the new deal id
+ * @param auditDetails - the users audit details
+ * @returns void
+ */
+export const cloneFacilities = async (currentDealId: string, newDealId: ObjectId, auditDetails: AuditDetails): Promise<void> => {
+  const facilitiesCollection = 'facilities';
+  const collection = await mongoDbClient.getCollection(facilitiesCollection);
+
+  const existingFacilities = (await collection
+    .find({
+      dealId: { $eq: new ObjectId(currentDealId) },
+    })
+    .toArray()) as Partial<Facility>[];
+
+  if (!existingFacilities.length) {
+    return;
+  }
+
+  const clonedFacilities = existingFacilities.map((facility) => {
```

Comment:
Same as the deal level cloning, can this be in its own file/function please and mention "sanitise"?